src/MultiVariablePolynomial.java

```diff
@@ -0,0 +1,57 @@
+import java.util.LinkedList;
+public class MultiVariablePolynomial
+{
+    String str;
+    LinkedList<LinkedList<Integer>> list = new LinkedList<LinkedList<Integer>>();
+
+    MultiVariablePolynomial(String str)
+    {
+        this.str = str;
+    }
+
+    int calculateMaxDegree()
+    {
+        int max = Integer.MIN_VALUE;
+        processLinkedList();
+        for (LinkedList<Integer> term : list)
+        {
+            int sum = 0;
+            for (int value : term)
+            {
+                sum += value;
+            }
+            max = sum > max? sum: max;
+        }
+        return max;
+    }
+
+    void processLinkedList()
+    {
+        String[] str1 = str.split("[+-]"); //Regular expression
+        for (String value : str1)
+        {
+            LinkedList<Integer> list = new LinkedList<Integer>();
+            for (int loop_var = 0; loop_var < value.length(); loop_var++)
+            {
+                if (value.charAt(loop_var) == 'x' || value.charAt(loop_var) == 'y')
```

Comment:
Why you are assuming that variable could be only x and y? I think you should accept all types of the variable like a.....z,A...Z and so on

Reply:
resolved, now it can be any lowercase character